treewide: jdk: make it point to latest version

doronbehar opened the pull request:
###### Motivation for this change

I want to encourage contributors to use (or at least try to use) the latest openjdk when they add new derivations that need Java. This change strives to make `jdk` point to the latest `jdk`, while every other derivation that uses `jdk` without an explicit version suffix, now uses `jdk8` explicitly.

I'm still waiting for ofborg to tag this PR as "10.rebuild-linux: 0", that's why it's a draft.

See also https://discourse.nixos.org/t/nixpkgs-policy-regarding-libraries-available-in-multiple-versions/7026 .

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

nixos-discourse commented:
This pull request has been mentioned on **NixOS Discourse**. There might be relevant details there:

https://discourse.nixos.org/t/nixpkgs-policy-regarding-libraries-available-in-multiple-versions/7026/4

doronbehar commented:
I'm thinking that perhaps, it'll be better to make `jdk` point to the latest `openjdk`, but write a script that will make every derivation that has `jdk` in it's arguments, use `jdk8` explicitly. This way, the PR will be tagged as `0: rebuild-linux` and so it'll be still easy enough to merge this.

doronbehar committed:
treewide: Make jdk refer to the latest jdk.

Make every other derivation, use jdk8 explicitly, so attributes won't
appear as changed by ofborg.

doronbehar commented:
Closed in favor of https://github.com/NixOS/nixpkgs/pull/89731 .
